src/App/App.js

```diff
@@ -40,30 +41,60 @@ class App extends Component {
       .then(data => this.setState({... this.state, movies: data.movies }))
   }
 
+  findMovie = (id) => {
+    return this.state.fetchedMovies.find(movie => movie.id === id)
+  }
+
   viewMovie = (id) => {
-      getMovie(id)
-        .then(response => response.json())
-        .then(data => {
-          this.setState({...this.state, isMovieView: true, movie: {
-            id: data.movie.id,
-            title: data.movie.title,
-            posterPath: data.movie['poster_path'],
-            backdropPath: data.movie['backdrop_path'],
-            releaseDate: data.movie['release_date'],
-            overview: data.movie.overview,
-            genres: data.movie.genres,
-            budget: data.movie.budget,
-            revenue: data.movie.revenue,
-            runtime: data.movie.runtime,
-            tagline: data.movie.tagline,
-            averageRating: data.movie['average_rating'],
+    const movie = this.findMovie(id);
+    if(movie) {
```

Comment:
if move is stored, then the data from that is used to update the movie that should be displayed

Reply:
this is going to reduce so many fetch requests nice job!!!